apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -0,0 +1,74 @@
+import { TRPCError } from "@trpc/server"
+import { z } from "zod"
+
+import { protectedProcedure, router } from "~/server/trpc"
+import { db } from "../database"
+
+export const resourceRouter = router({
+  getMetadataById: protectedProcedure
+    .input(
+      z.object({
+        resourceId: z.coerce.number().min(1),
+      }),
+    )
+    .query(async ({ input: { resourceId } }) => {
+      const resource = await db
+        .selectFrom("Resource")
+        .where("Resource.id", "=", resourceId)
+        .select(["permalink", "Resource.id", "Resource.title"])
+        .executeTakeFirst()
+
+      if (!resource) {
+        throw new TRPCError({ code: "NOT_FOUND" })
+      }
+
+      return resource
+    }),
+
+  getChildrenOf: protectedProcedure
+    .input(
+      z.object({
+        resourceId: z.union([z.coerce.number().min(1), z.null()]),
+      }),
+    )
+    .query(async ({ input: { resourceId } }) => {
+      let query = db
+        .selectFrom("Resource")
+        .select(["title", "permalink", "type", "id"])
+        .orderBy("type", "asc")
+        .orderBy("title", "asc")
+
+      if (resourceId === null) {
```

Comment:
nit: could use `!resourceId`

Reply:
not doing - `null` is meant to actually be the `null` value not any falsy value. ie, an empty string shouldn't be treated the same as `null` here hence the explicit check

Reply:
yea. Also might have resourceId of 0, etc seeing how it's a big int, may lead to edge cases (though i'm sure they start from 1, but semantically wrong)